src/main/java/org/broadinstitute/hellbender/utils/collections/NestedIntegerArray.java

```diff
@@ -83,6 +83,89 @@ public T get(final int... keys) {
     }
 
     /**
+     * Specialized version of get for 1 parameter.
+     * Varagrs have a large cost because the arg array is allocated everytime.
+     * Using a specialized method eliminates that performance problem.
+     */
+    @SuppressWarnings("unchecked")
+    public T get1(final int key0) {
+        Object[] myData = data;
+        return (T)myData[key0];
+    }
+
+    /**
+     * Specialized version of get for 2 parameters.
+     * Varagrs have a large cost because the arg array is allocated everytime.
+     * Using a specialized method eliminates that performance problem.
+     */
+    @SuppressWarnings("unchecked")
+    public T get2(final int key0, final int key1) {
+        if ( key0 >= dimensions[0] ) { return null; }
+
+        Object[] myData = data;
+        myData = (Object[])myData[key0];
+        if ( myData == null ) { return null; }
+
+        return (T)myData[key1];
```

Comment:
There's a lot of code duplication here and below.  Why not implement this method as:
final Object[] myData = get1(key0);
if ( myData == null ) { return null; }
return (T)myData[key1];

And we should be able to do something similar below, no?